src/SMU-HP_4141B/main.py

```diff
@@ -0,0 +1,216 @@
+# This Device Class is published under the terms of the MIT License.
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+# 
+# Copyright (c) 2024 SweepMe! GmbH (sweep-me.net)
+# 
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! driver
+# * Module: SMU
+# * Instrument: HP 4141B
+
+
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+
+class Device(EmptyDevice):
+
+    def __init__(self):
+
+        EmptyDevice.__init__(self)
+
+        self.variables = ["Voltage", "Current"]
+        self.units = ["V", "A"]
+        self.plottype = [True, True]  # True to plot data
+        self.savetype = [True, True]  # True to save data
+
+        self.port_manager = True
+        self.port_types = ["GPIB"]
+        self.port_properties = {
+            "timeout": 3,
+        }
+
+        self.current_ranges = {
+            "Auto": "0",
+            "1 nA limited auto": "1",
+            "10 nA limited auto": "2",
+            "100 nA limited auto": "3",
+            "1 μA limited auto": "4",
+            "10 μA limited auto": "5",
+            "100 μA limited auto": "6",
+            "1 mA limited auto": "7",
+            "10 mA limited auto": "8",
+            "100 mA limited auto": "9",
+        }
+
+    def set_GUIparameter(self):
+
+        GUIparameter = {
+            "SweepMode": ["Voltage in V", "Current in A"],
+            "Channel": ["CH1", "CH2", "CH3", "CH4"],
+            "RouteOut": ["Rear"],
+            # "Speed": ["Fast", "Medium", "Slow"],
+            "Range": list(self.current_ranges.keys()),
+            "Compliance": 100e-6,
+            "Average": 1,
+        }
+
+        return GUIparameter
+
+    def get_GUIparameter(self, parameter={}):
+
+        self.device = parameter['Device']
+        self.four_wire = parameter['4wire']
```

Comment:
This is not a parameter in set_GUIparameter. Same for 42B driver

Reply:
changed it